controllers/gce/controller/controller.go

```diff
@@ -342,6 +349,39 @@ func (lbc *LoadBalancerController) sync(key string) (err error) {
 	return syncError
 }
 
+func (lbc *LoadBalancerController) syncMultiClusterIngress(ing *extensions.Ingress, nodeNames []string) error {
```

Comment:
There is duplicate logic here with Checkpoint. Can you refactor Checkpoint to avoid having dupplicate logic? 

Currently, `CreateInstanceGroup` is done as part of syncing backend services. Extract `create and sync instance group`, and set the annotation, if `gce-multi`, then return early, otherwise, continue for the rest.

This requires more changes to existing logic though. I will let @nicksardo to have more input for this.

Reply:
yeah I dont want to call the whole Checkpoint method just to sync instance groups. I think its better to call that specific method directly here. This also ensures that the existing ingress code is independent and unchanged.